src/cls/rgw/cls_rgw.cc

```diff
@@ -1178,6 +1301,15 @@ int rgw_bucket_complete_op(cls_method_context_t hctx, bufferlist *in, bufferlist
 		   __func__, escape_str(idx).c_str(), rc);
       return rc;
     }
+    if (header.resharding_in_logrecord()) {
```

Comment:
nit: since this code block appears at least six separate times, I wonder if we could use a function or lambda for it.

Reply:
Fine.